pkg/util/validation/errors.go

```diff
@@ -100,8 +100,15 @@ func newTooManyLabelsError(series []mimirpb.LabelAdapter, limit int) ValidationE
 }
 
 func (e tooManyLabelsError) Error() string {
+	metric := mimirpb.FromLabelAdaptersToMetric(e.series).String()
+	ellipsis := ""
+
+	if len(metric) > 200 {
```

Comment:
This is not correct. When formatting strings using `Sprintf`, "%.200s" means 200 runes. But here we check for 200 bytes. Runes outside of ASCII range will use more bytes, so we will add ellipsis even when entire metric was printed.

Reply:
Good catch, fixed in ceade5c